tools/azsdk-cli/docs/specs/7-repo-tooling-contract.md

```diff
@@ -0,0 +1,342 @@
+Without a standardized approach, it is difficult to reason about potential breaking changes when `azsdk`
+is tightly coupled to the call pattern of scripts in other repositories.
+
+This spec builds upon the ideas outlined in [6-package-updater-spec.md](https://github.com/Azure/azure-sdk-tools/blob/cc5bae3dd44b02c01ebb8d7546fdbeb0838b665d/tools/azsdk-cli/docs/specs/6-package-updater.spec.md#proposed-approaches-for-the-intergration-between-language-specific-logic-and-tools) and the [preceding discussion](https://github.com/Azure/azure-sdk-tools/pull/12563#discussion_r2449474956). It seemed easier to focus that discussion into this separate spec.
+
+### Current State
+
+- Scripts can be referenced directly in `azsdk` by a hardcoded file path in the process helper [example](https://github.com/Azure/azure-sdk-tools/blob/826bd97dc1580e65b82ac7e4c2c4cd35a1ff8c3a/tools/azsdk-cli/Azure.Sdk.Tools.Cli/Services/Languages/DotNetLanguageSpecificChecks.cs#L54)
+- There are no CI tests in the target repositories to prevent breaking changes being made to referenced scripts.
+- There is no formal versioning/dependency between `azsdk` and the repository state.
+
+### Why This Matters
+
+There is years worth of hardened automation that has been driving the engineering system in the language repositories.
+As the `azsdk` tool is being developed it offers opportunities to streamline some of this automation and also make it accessible
+to AI agents.
+It seems likely that existing repository scripts will always exist in some capacity (see below), so a formalized approach to referencing
+these scripts will have the following benefits:
+
+- Improved velocity - tactical script changes can be made with confidence while not requiring an `azsdk` release.
+  - As `azsdk` matures and gains usage/impact, the validations/requirements to release a new version will increase.
+  - Small, language-specific changes can be made without needing to understand `azsdk` development.
+  - Breakglass changes can be made to resolve urgent issues (unblocking releases, security issues, etc.)
+- Easier migration of scripts to `azsdk`.
+  - It will take time to centralize automation into `azsdk`. There needs to be a bridge between `azsdk` and existing scripts that can support a gradual migration while maintaining reliability.
+- New automation experiments can be made in language repositories, even if the desired end state is a generic interface in `azsdk`.
+- Language specific edge cases can exist without cluttering the cli/mcp tool list.
+- Depending on calling a script (i.e. Export-Apis.ps1) or command line (msbuild /p:ExportApis) that have a set of parameters unique to our repo will be fragile and make versioning `azsdk` with the source changes in the repo very difficult. It will also make this tool very difficult to use in other repo contexts (i.e. the openai or mcp repos).
```

Comment:
Also export-api is a terrible example because it hasn't had a change in parameters i.e. contract since May 2021